sample/src/main/java/com/davidecirillo/multichoicesample/sampleCustomView/SampleCustomActivity.java

```diff
@@ -33,33 +28,26 @@ protected int setActivityIdentifier() {
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
         setUpMultiChoiceRecyclerView();
     }
 
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        ((MultiChoiceAdapter) multiChoiceRecyclerView.getAdapter()).onSaveInstanceState(outState);
```

Comment:
I'd like it more, when we have the adapter stored in a field (with the correct type) so no cast is needed but it should be fine this way, too (same applies to the other classes as well).